cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/FloatDelta.qll

```diff
@@ -27,3 +28,44 @@ module FloatDelta implements DeltaSig {
       )
   }
 }
+
+module FloatOverflow implements OverflowSig<FloatDelta> {
+  predicate semExprDoesntOverflow(boolean positively, SemExpr expr) {
+    exists(float lb, float ub, float delta |
+      typeBounds(expr.getSemType(), lb, ub) and 
+      ConstantStage::initialBounded(expr, any(ConstantBounds::SemZeroBound b), delta, positively.booleanNot(), _, _, _)
+    |
+      positively = true and delta < ub
+      or
+      positively = false and delta > lb
+    )
```

Comment:
Just checking my understanding here: We're finding the minimum and maximum values from the the of `expr`, and then we're comparing those values with the bounds we can obtain from the constant stage.

I'm not sure I understand the logic, though. If we're trying to deduce that `expr` doesn't overflow _positively_, we're checking whether `delta` (which is the _lower bound_ of `expr`) is less than the maximum value of the type? Shouldn't this be the other way around? Or have I confused myself?